tests/panels/test_history.py

```diff
@@ -28,13 +28,21 @@ def test_post(self):
         self.assertEqual(data["foo"], "bar")
 
     def test_post_json(self):
-        self.request = rf.post(
-            "/", data={"foo": "bar"}, content_type="application/json"
-        )
-        response = self.panel.process_request(self.request)
-        self.panel.generate_stats(self.request, response)
-        data = self.panel.get_stats()["data"]
-        self.assertEqual(data["foo"], "bar")
+        for data, expected_stats_data in (
+            ({"foo": "bar"}, {"foo": "bar"}),
+            ("", {}),
+        ):
+            with self.subTest(data=data):
+                self.request = rf.post(
+                    "/",
+                    data=data,
+                    content_type="application/json",
+                    CONTENT_TYPE="application/json",  # Force django test client to add the content-type even if no data
```

Comment:
Are both lines necessary? Or is it possible to remove one of the content_type parameters? (I didn't check what they do, do maybe it isn't a very smart question.)

Reply:
I think the second is to force the header option.

Reply:
Ah yes I see. The first makes Django automatically encode the `data` as JSON, the second adds the header even if `data` is falsy. 

It's certainly convenient that Django automatically encodes the data if the content_type has some particular value. Sometimes the interdependence of different arguments just isn't obvious (enough) to me.

Reply:
Yes, unfortunately, this looks more like a trick, the django client don't add the `content_type` if there is no data, so I add manually (extra kwargs which are not known by the test client are considered as extra headers).